lib/type.g

```diff
@@ -314,14 +367,52 @@ DeclareRepresentation( "IsAtomicPositionalObjectRep",
 ##  <Filt Name="IsAttributeStoringRep" Arg='obj' Type='Representation'/>
 ##
 ##  <Description>
-##  Objects in this representation have default  methods to get the values of
-##  stored  attributes  and -if they  are immutable-  to store the  values of
-##  attributes after their computation.
+##  Objects in this representation have default methods to get stored values
+##  of attributes and &ndash;if they are immutable&ndash; to store attribute
+##  values automatically once they have been computed.
+##  <Index>system getter</Index>
+##  <Index>system setter</Index>
+##  (These methods are called the <Q>system getter</Q> and the
+##  <Q>system setter</Q> of the attribute, respectively.)
+##  <P/>
+##  As a consequence,
+##  for immutable objects in <Ref Filt="IsAttributeStoringRep"/>,
+##  subsequent calls to an attribute will return the <E>same</E> object.
+##  <P/>
+##  <E>Mutable</E> objects in <Ref Filt="IsAttributeStoringRep"/>
+##  are allowed, but attribute values are not stored automatically in them.
+##  Such objects are useful because they may later be made immutable using
+##  <Ref Func="MakeImmutable"/>, at which point they will start storing
+##  all attribute values.
+##  <P/>
+##  Note that one can force an attribute value to be stored in a mutable
+##  object in <Ref Filt="IsAttributeStoringRep"/>,
+##  by explicitly calling the attribute setter.
+##  This feature should be used with care.
+##  For example, think of a mutable matrix whose rank or trace gets stored,
+##  and the values later become wrong when somebody changes the matrix
+##  entries.
 ##  <P/>
+##  <Example><![CDATA[
+##  gap> g:= Group( (1,2)(3,4), (1,3)(2,4) );;
+##  gap> IsAttributeStoringRep( g );
+##  true
+##  gap> HasSize( g );  Size( g );  HasSize( g );
+##  false
+##  4
+##  true
+##  gap> r:= 7/4;;
+##  gap> IsAttributeStoringRep( r );
+##  false
+##  gap> Int( r );  HasInt( r );
+##  1
+##  false
+##  ]]></Example>
+##  <!--
 ##  The name of the  component that holds  the value of  an attribute is  the
 ##  name of the attribute, with the first letter turned to lower case.
-##  <!-- This will be changed eventually, in order to avoid conflicts between-->
-##  <!-- ordinary components and components corresponding to attributes.-->
+##  This will be changed eventually, in order to avoid conflicts between
+##  ordinary components and components corresponding to attributes.-->
```

Comment:
Perhaps we should just remove those last 4-5 lines?

Reply:
Yes.
Note that the rule for the component names had not been commented out before, but still it was not part of the documentation since the GAPDoc "chunk" was not requested by any of the XML files.